### PR TITLE
Enable CI failures for linting

### PR DESCRIPTION
**Pull request by bmoffatt**
WIP

### CODE REVIEW

**Comment by philtay** [comment]
@bmoffatt I suggest to use GolangCI-Lint instead of gometalinter. https://github.com/golangci/golangci-lint

File: .travis.yml [hunk 1]
```diff
@@ -34,13 +35,11 @@ notifications:
...
 before_script:
   - PKGS=$(go list ./...)
-  - LINT_PKGS=$(go list ./... | grep -Ev 'aws-lambda-go/lambda')
 
 script:
   - diff -u <(echo -n) <(gofmt -d ./)                              # Fail if a .go file hasn't been formatted with gofmt
   - goverage -v -covermode=atomic -coverprofile=coverage.txt $PKGS # Run all tests with coverage 
-  - go vet -v ./...                                                # static analyisis
-  - golint $LINT_PKGS                                              # lint - ignore failures for now
+  - gometalinter --vendor --disable-all --enable=vet --enable=golint ./...
```
Comment (oleg-balunenko) [inline .travis.yml]: DEPRECATED: Use https://github.com/golangci/golangci-lint
see notes from the author of https://github.com/alecthomas/gometalinter

File: .travis.yml [hunk 2]
```diff
@@ -34,13 +35,11 @@ notifications:
 
 before_script:
   - PKGS=$(go list ./...)
-  - LINT_PKGS=$(go list ./... | grep -Ev 'aws-lambda-go/lambda')
 
 script:
   - diff -u <(echo -n) <(gofmt -d ./)                              # Fail if a .go file hasn't been formatted with gofmt
   - goverage -v -covermode=atomic -coverprofile=coverage.txt $PKGS # Run all tests with coverage 
-  - go vet -v ./...                                                # static analyisis
```
[the remark this answers is not in the document]
Comment (bmoffatt) [reply .travis.yml]: Will leave it in!

**Comment by bmoffatt** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: aws-lambda-go-github-ci
* Commit ID: fca96002ee7c2700ce2273277902291019195b12
* Result: SUCCEEDED
* [Build Logs](https://yxbry9e1t8.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=88c6b7b8-db17-47f1-b847-29edd84e9d4e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**